test/common/utils_concurrency_limit.h

```diff
@@ -309,6 +309,36 @@ void decrease_thread_priority() {
 #endif
 }
 
+#if __unix__
+class increased_priority_guard {
+public:
+    increased_priority_guard() : m_backup(get_current_schedparam()) {
+        increase_thread_priority();
```

Comment:
I have checked the oneTBB repo and it seems like `increase_thread_priority` only used in test_task in the places you changed. And `decrease_thread_priority` is never used. 
May be it makes sense to remove `decrease_thread_priority` and move `increase_thread_priority` implementation into the guard to prevent its usage outside of the guard?

Reply:
Yeah, makes sense. Thank you.